### ISSUE TITLE
Apex Runtime Error: Broken pipe

### ISSUE THREAD

**Issue by SwagJ** [NVIDIA/apex#291]
Hi, 

The problem is that when I run the code it gives a broken pipe runtime error
During the installation of apex, i got the same warning for a couple of times: "cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++"
Although in the end, it showed that it's been successfully installed, will this possibly be the problem with broken pipe issue? If not, could you please share some ideas on solution? Thank you very much!

best,

**Comment by ptrblck** [comment]
Hi @SwagJ,

the warning is most likely unrelated to your problem. I think the broken pipe error might be a Python/multiproceesing issue. 
Are you using multiple workers in your `DataLoader`? If so, could you try to run your code with `num_workers=0` and check if it's working?
If that's the case, you might need to add the if-clause protection as described [here](https://pytorch.org/docs/stable/notes/windows.html#multiprocessing-error-broken-pipe).

Let me know, if that helps.

**Comment by SwagJ** [comment]
Hi @ptrblck ,
Thank you very much for your reply! Previously, the num_workers is set to 8, now it is 0. Indeed, it fixed the problem with broken pip, but it raised error with address already existed. And I also, tried to add if-clause protection, but it doesn't work somehow. The command I used to run the code is the following:
export NGPUS=8
python -m torch.distributed.launch --nproc_per_node=$NGPUS train.py
Thank you very much!

**Comment by ptrblck** [comment]
In that case the broken pipe error might just have masked the real issue with the distributed launch. 
Could you check, if old processes have not been killed properly and are still using the port as described in [this post](https://discuss.pytorch.org/t/multiprocessing-failed-with-torch-distributed-launch-module/33056)?

**Comment by mcarilli** [comment]
I find the following command useful to clean up after distributed launches:
```
ps aux | grep python | awk '{ print $2 }' | xargs kill -9
```
This will kill all Python processes, including dataloader subprocesses.

**Comment by mcarilli** [comment]
Also, using 8 dataloader subprocesses per process is a lot in our experience.  We typically don't use more than 4, and we don't see a benefit beyond that.  Of course this depends on the number of cores on your system.  If you have a lot of free cores, it might be beneficial to increase the number of dataloader subprocesses.

Has this issue been resolved?  Can we close it?

**Comment by SwagJ** [comment]
Yeah, it has been fixed, I think you are right. That is how I fix my problem. Thank you very much!
